File: getToken.js
// getToken.js
const AWS = require('aws-sdk');

const USER_POOL_ID = 'us-east-1_7QZ7Y6Gbl';
const CLIENT_ID = '6gobbpage9af3nd7ahm3lchkct';
const REGION = 'us-east-1';
const username = 'testuser';
const password = 'TestPassword123!';

const cognito = new AWS.CognitoIdentityServiceProvider({ region: REGION });

const params = {
    AuthFlow: 'USER_PASSWORD_AUTH',
    ClientId: CLIENT_ID,
    AuthParameters: {
        USERNAME: username,
        PASSWORD: password,
    },
};

cognito.initiateAuth(params, (err, data) => {
    if (err) {
        console.error("Authentication failed:", err);
        process.exit(1);
    } else {
        console.log(data.AuthenticationResult.IdToken);
    }
});

File: getToken.py
import json
import requests

url = "https://cognito-idp.us-east-1.amazonaws.com/"

headers = {
    "Content-Type": "application/x-amz-json-1.1",
    "X-Amz-Target": "AWSCognitoIdentityProviderService.InitiateAuth",
}

data = {
    "AuthFlow": "USER_PASSWORD_AUTH",
    "ClientId": "6gobbpage9af3nd7ahm3lchkct",
    "AuthParameters": {
        "USERNAME": "testuser",
        "PASSWORD": "TestPassword123!",
    },
}

response = requests.post(url, headers=headers, data=json.dumps(data))

if response.status_code == 200:
    print(response.json()["AuthenticationResult"]["IdToken"])
else:
    print(f"Authentication failed: {response.text}")
    exit(1)
